### PR TITLE
Adds BmcReset methods and few fixes for ASRR firmware installed version checks

### PR DESCRIPTION
**Pull request by joelrebel**
* Checklist

- [ ] Tests added
- [x] Similar commits squashed

What does this PR implement/change/remove?
  - builds on changes in #261 
  - fixes the redfish `InventoryGetter` implementation - method renamed
  - implements the `BmcResetter` interface for ASRR, Redfish
  - Firmware install status lookup to return state `FirmwareInstallUnknown`, when the BMC returns an empty version string
  - A few other minor fixes

The HW vendor this change applies to
  - ASRR
  - Redfish (provider)

Description for changelog/release notes
 - Implement `BmcResetter` interface for ASRR, Redfish

### CODE REVIEW

File: providers/asrockrack/firmware.go
Comment (micahhausler) [inline providers/asrockrack/firmware.go:128]: Waiting on removal for approval

File: providers/asrockrack/firmware.go
Comment (micahhausler) [inline providers/asrockrack/firmware.go:145]: Maybe log the error in a structured method?
```suggestion
		a.log.V(3).Error(err, "failed bmc query for install progress ")
```

File: providers/asrockrack/firmware.go
Comment (micahhausler) [inline providers/asrockrack/firmware.go:155]: Since you're emitting structured logs, maybe put the state in a json field?
```suggestion
			a.log.V(3).WithValues("state", pgorgess.State).Info("bmc returned unknown flash progress state")
```

**Review by micahhausler — CHANGES_REQUESTED** [review]
Just a few minor changes, thanks!

File: providers/asrockrack/firmware.go
Comment (joelrebel) [reply providers/asrockrack/firmware.go]: thanks I didn't realise I could log `WithValues()` in logr

File: examples/v1/firmware/firmware.go
Comment (micahhausler) [inline examples/v1/firmware/firmware.go:21]: Want to copy/paste the `flag` imports from the other examples?

File: examples/v1/firmware/firmware.go
Comment (micahhausler) [inline examples/v1/firmware/firmware.go:42]: want to make this a flag-provided value?

File: examples/v1/firmware/firmware.go
Comment (micahhausler) [inline examples/v1/firmware/firmware.go:54]: Why remove the status check?

**Review by micahhausler — COMMENTED** [review]
LGTM, unless you want to add flags to the example

**Review by micahhausler — APPROVED** [review]

File: examples/v1/firmware/firmware.go
Comment (joelrebel) [reply examples/v1/firmware/firmware.go:21]: will include the `flag` imports at rebase in the parent branch.

**Pull request merged**
